cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -294,10 +304,36 @@ class HydratorPlusPlusTopPanelCtrl {
     seconds = seconds < 10 ? '0' + seconds : seconds;
     minutes = minutes < 10 ? '0' + minutes : minutes;
 
+    this.setDisplayDuration(minutes, seconds);
+  }
+
+  setDisplayDuration(minutes, seconds) {
     this.displayDuration = {
-      minutes: minutes,
-      seconds: seconds
-    };
+      minutes: !minutes ? '--' : minutes,
+      seconds: !seconds ? '--' : seconds,
+    }
+  }
+
+  updateTimerLabelAndTitle(res) {
+    // set default
+    if (!res) {
+      this.timerLabel = 'Duration';
+      this.queueStatus = '';
+      return;
+    }
+
+    const { WAITING, ACQUIRED, INIT, RUNNING } = window.CaskCommon.PREVIEW_STATUS;
+    if (res.status === WAITING) {
+      const runsAheadInQueue = res.positionInWaitingQueue;
+      this.queueStatus = `${runsAheadInQueue} ${runsAheadInQueue === 1? 'run' : 'runs'} ahead in queue`;
+      this.timerLabel = `${runsAheadInQueue} Pending`;
+    } else if ([ ACQUIRED, INIT, RUNNING ].includes(res.status) && this.loadingLabel !== 'Stopping') {
+      this.timerLabel = 'Running';
```

Comment:
Could we use `PREVIEW_STATUS` for `'Running'` and `'Duration'`?

Reply:
That is an interesting idea! I'm not sure it makes sense to use PREVIEW_STATUS for the timer labels, since they sometimes don't correlate and "Duration" is not a preview status.

Reply:
Alrighty!